### DISCUSSION TITLE
Docker no longer launches on 1.69.x

### DISCUSSION THREAD

**Discussion by mlschuh** [gitpod-io/openvscode-server#392] [General] [answered]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: Working on 1.68.x, not functional on 1.69.x
- OS Version: Tried on docker 19.03.4 on Ubuntu & docker 20.10.3 on Alpine

Steps to Reproduce:

1. Run the supplied `docker run -it --init -p 3000:3000 -v "$(pwd):/home/workspace:cached" gitpod/openvscode-server:1.68.1`, launches properly.
2. Run the supplied `docker run -it --init -p 3000:3000 -v "$(pwd):/home/workspace:cached" gitpod/openvscode-server:1.69.2`, throws the following error:

```
/home/.openvscode-server/node[12]: ../src/node_platform.cc:61:std::unique_ptr<long unsigned int> node::WorkerThreadsTaskRunner::DelayedTaskScheduler::Start(): Assertion `(0) == (uv_thread_create(t.get(), start_thread, this))' failed.
 1: 0xb00e10 node::Abort() [/home/.openvscode-server/node]
 2: 0xb00e8e  [/home/.openvscode-server/node]
 3: 0xb7045e  [/home/.openvscode-server/node]
 4: 0xb70526 node::NodePlatform::NodePlatform(int, v8::TracingController*) [/home/.openvscode-server/node]
 5: 0xac43f9 node::InitializeOncePerProcess(int, char**, node::InitializationSettingsFlags) [/home/.openvscode-server/node]
 6: 0xac4696 node::Start(int, char**) [/home/.openvscode-server/node]
 7: 0x7fbaa9391d90  [/lib/x86_64-linux-gnu/libc.so.6]
 8: 0x7fbaa9391e40 __libc_start_main [/lib/x86_64-linux-gnu/libc.so.6]
 9: 0xa3bf3c  [/home/.openvscode-server/node]
Aborted (core dumped)
```

The timing is across a minor release with many updates so I'm not quite sure where to look but this issue (https://github.com/gitpod-io/openvscode-server/issues/377) caught my eye as directly relating to the docker image and being in that exact timeframe.

**Comment by jeanp413** [comment, marked as answer]
Hi @mlschuh you need to upgrade your docker version, I have `Docker version 20.10.17, build 100c701` and I don't get this error, found this issue releated  [[1](https://github.com/nodejs/node/issues/43064)]

Also from [this](https://askubuntu.com/questions/1405417/20-04-vs-22-04-inside-docker-with-a-16-04-host-thread-start-failures)  you can workaround it by adding `--security-opt seccomp=unconfined` to the docker command

See also https://askubuntu.com/a/1415118 and https://github.com/adoptium/containers/issues/215#issuecomment-1142046045

**Reply by homerokzam** [reply]
I am with this version of Docker and I have the same problem.
I solved it by physically mapping the volume (-v "/root/vscode:/home/workspace:cached"), and adding the root user (--user=$UID).
